Backend/Backend/Services/PodcastService.cs

```diff
@@ -571,5 +571,10 @@ public async Task<string> GetEpisodeThumbnailNameAsync(Guid episodeId)
         return episode.Thumbnail;
     }
 
+    public async Task<UserEpisodeInteraction?> GetUserEpisodeInteraction(User user, Episode episode)
```

Comment:
What if a user has many interactions with the same episode? We would want to return the latest interaction (only if the audio has not been fully listened).

Reply:
There shouldn't be. 1 interaction per episode per user. Otherwise it is a bug. I think we need to add restriction in DB as well